pkg/agent/cniserver/ipam/antrea_ipam_controller.go

```diff
@@ -148,6 +148,20 @@ func (c *AntreaIPAMController) getIPPoolsByPod(namespace, name string) ([]string
 		}
 		return nil, nil, nil, err
 	}
+
+	annotations, exists := pod.Annotations[AntreaIPAMAnnotationKey]
+	if !exists {
+		// Find IPPool by Namespace
+		ns, err := c.namespaceLister.Get(namespace)
+		if err != nil {
+			return nil, nil, nil, nil
+		}
+		annotations, exists = ns.Annotations[AntreaIPAMAnnotationKey]
+		if !exists {
+			return nil, nil, nil, nil
+		}
+	}
+
```

Comment:
If we move this here, some behavior will be changed.

1. Pod with IPAnnotaion, Namespace with IPPoolAnnotation. This case will be broken.
2. StatefulSet Pod, Namespace with IPPoolAnnotation. The StatefulSetOwner will be no longer recorded on IPPoolStatus.

Reply:
Please ignore this.